### ISSUE TITLE
Document how to wrap Refile application in authentication middleware.

### ISSUE THREAD

**Issue by jnicklas** [refile/refile#352]
See also #338. For people who are wary of unauthenticated requests to the Refile application, we should document how to create a middleware which wraps the application in an authentication scheme.

**Comment by fabn** [comment]
I was reading [your answer](https://github.com/refile/refile/issues/338#issuecomment-138913611) in #338 and I don't think that a middleware is the right choice here. Let me explain the reasons:

* Middleware code is executed in every request even unrelated ones
* Doing that in middleware means parse the requested url, check if belongs to Refile::App, check for authentication/authorization, halt if user is not allowed to do the upload.
* You will only have access to `env` and potentially nothing else.

Doing the authentication in Refile::App solve all the above issues and when used in rails you'll have access to `env['action_dispatch.cookies']`, `env['warden']` and all the goodies available in sinatra with no need to parse the env hash (very error prone).

Have you seen my proposal in my [latest answer](https://github.com/refile/refile/issues/338#issuecomment-138493112)? Would you be interested in a pull request attempt?

**Comment by jnicklas** [comment]
Just realized that there's a way easier option. Since `Refile::App` is a Sinatra application you can do something like this:

``` ruby
Refile::App.before do
  if request.post?
    halt 403 unless env["warden"].whatever?
  end
end
```

This gives you access to the request and anything else inside the app.

P.S.: You could also do everything you stated in your middleware, but this is easier anyway, so whatever.

**Comment by fabn** [comment]
That's way better of everything else. In this way one can do everything he needs.

Thanks for the hint.

**Comment by jnicklas** [comment]
Closed by 68987fe98f1432dc9e6fa532940331f77193cfc8

Accidentally put the wrong issue number in the commit.
